packages/core/src/indexing/local-index-api.ts

```diff
@@ -17,27 +19,34 @@ import { Model } from '@ceramicnetwork/stream-model'
 import { ISyncQueryApi } from '../sync/interfaces.js'
 
 /**
- * Takes a Model StreamID, loads it, and returns the IndexModelArgs necessary to prepare the
+ * Takes a ModelData, loads it, and returns the IndexModelArgs necessary to prepare the
  * database for indexing that model.
  */
 async function _getIndexModelArgs(
-  modelStreamId: StreamID,
-  repository: Repository
+  req: ModelData,
+  repository: Repository,
+  databaseIndexApi?: DatabaseIndexApi
 ): Promise<IndexModelArgs> {
+  const modelStreamId = req.streamID
   if (modelStreamId.type != Model.STREAM_TYPE_ID && !modelStreamId.equals(Model.MODEL)) {
     throw new Error(`Cannot index ${modelStreamId.toString()}, it is not a Model StreamID`)
   }
 
+  const opts: IndexModelArgs = {
+    model: modelStreamId,
+  }
+
   if (modelStreamId.type == Model.STREAM_TYPE_ID) {
     const modelState = await repository.load(modelStreamId, {})
     const content = modelState.state.next?.content ?? modelState.state.content
     Model.assertVersionValid(content, 'major')
     if (content.relations) {
-      return { model: modelStreamId, relations: content.relations }
+      opts.relations = content.relations
     }
+    opts.indices = req.indices ?? (await databaseIndexApi?.getFieldsIndex(modelStreamId))
```

Comment:
I know I asked this like 3 times on the earlier version of this PR but I just still don't get why this is necessary.  In the follow-up PR can you try adding a comment here explaining why this is necessary, and if it's still not clear to me maybe we can hop on quick call to discuss?